searx/infopage/__init__.py

```diff
@@ -116,20 +116,20 @@ def __init__(
         self, page_class: typing.Optional[typing.Type[InfoPage]] = None, info_folder: typing.Optional[str] = None
     ):
         self.page_class = page_class or InfoPage
-        self.CACHE: typing.Dict[tuple, typing.Optional[InfoPage]] = {}
-
-        # future: could be set from settings.xml
-
         self.folder: str = info_folder or _INFO_FOLDER
         """location of the Markdwon files"""
 
+        self.CACHE: typing.Dict[tuple, typing.Optional[InfoPage]] = {}
+
         self.locale_default: str = 'en'
         """default language"""
 
-        self.locales: typing.List = [locale for locale in os.listdir(_INFO_FOLDER) if locale in LOCALE_NAMES]
+        self.locales: typing.List[str] = [
+            locale.replace('_', '-') for locale in os.listdir(_INFO_FOLDER) if locale.replace('_', '-') in LOCALE_NAMES
```

Comment:
* LOCALE_NAMES  contains names with dash
* The locale names in the infopage directory are going to use underscore (like in the translations directory)